app/User.php

```diff
@@ -244,6 +244,49 @@ public function getGreetingAndName()
         return $this->getGreeting() . ' ' . $this->getFullName();
     }
 
+    /**
+     * Get the user's record by Billet ID. For use in the retrieval of
+     * Space Lords and heads of branches, where a single user holds the
+     * specified billet.
+     *
+     * @param string $billet_id
+     *
+     * @return User|null
+     */
+    public static function getByBilletId(string $billet_id)
+    {
+        $billet_user = null;
+
+        $billet = Billet::where('_id', $billet_id)->first();
+
+        if (isset($billet) && isset($billet['billet_name'])) {
```

Comment:
@cinnion Check class type of $billet

Yes, I know it hasn't been done before, but I've learned a shit ton more about PHP and best practices since this was first written.  We may not be able to refactor, but we can do things right going forward :-)